buildspec.yml

```diff
@@ -0,0 +1,135 @@
+version: 0.2
+
+env:
+  variables:
+    FRAMEWORK_VERSION: '1.3.0'
+    CPU_FRAMEWORK_BINARY: 'https://files.pythonhosted.org/packages/71/64/49c5125befd5e0f0e17f115d55cb78080adacbead9d19f253afd0157656a/mxnet-1.3.0.post0-py2.py3-none-manylinux1_x86_64.whl'
+    CPU_PY_VERSION: '2'
+    CPU_INSTANCE_TYPE: 'ml.c4.xlarge'
+    GPU_FRAMEWORK_BINARY: 'https://files.pythonhosted.org/packages/52/8b/3f9cfe199e592b0723617e5d1919b83022f05cc359bd1885f7d1e9ce4758/mxnet_cu90mkl-1.3.0.post0-py2.py3-none-manylinux1_x86_64.whl'
+    GPU_PY_VERSION: '3'
+    GPU_INSTANCE_TYPE: 'ml.p2.xlarge'
+    ECR_REPO: 'sagemaker-test'
+    GITHUB_REPO: 'sagemaker-mxnet-container'
+    SETUP_FILE: 'setup_cmds.sh'
+    SETUP_CMDS: '#!/bin/bash\npip install --upgrade pip\npip install -U -e .\npip install -U -e .[test]'
+
+
+phases:
+  pre_build:
+    commands:
+      - start-dockerd
+      - cp /usr/local/bin/remote_test.py .
+      - |
+        ACCOUNT=$(aws sts get-caller-identity --query 'Account' --output text)
+        PREPROD_IMAGE="$ACCOUNT.dkr.ecr.$AWS_DEFAULT_REGION.amazonaws.com/$ECR_REPO"
+      # keep ssh connection alive when communicating with remote ec2 server during integ test
+      # largest connection idle time allowed: 30 seconds * 60 attempts = 30 minutes
+      - |
+        echo '  ServerAliveInterval 30' >> ~/.ssh/config
+        echo '  ServerAliveCountMax 60' >> ~/.ssh/config
+
+
+  build:
+    commands:
+      # install
+      - pip3 install -U -e .
+      - pip3 install -U -e .[test]
+
+      # run unit tests
+      - pytest test/unit
+
+      # Create tensorflow-container pip archive
+      - |
+        build_dir="docker/$FRAMEWORK_VERSION/final"
+        build_id="$(echo $CODEBUILD_BUILD_ID | sed -e 's/:/-/g')"
+        python3 setup.py sdist
+        # tar_name will be something like: sagemaker_mxnet_container-1.0.tar.gz
+        tar_name=$(ls dist)
+        cp dist/$tar_name $build_dir
+
+      # build cpu image
+      - |
+        cpu_dockerfile="Dockerfile.cpu"
+
+        # Download framework binary
+        cpu_fw_binary=$(basename $CPU_FRAMEWORK_BINARY)
+        wget -O $build_dir/$cpu_fw_binary $CPU_FRAMEWORK_BINARY
+
+        CPU_TAG="$FRAMEWORK_VERSION-cpu-py$CPU_PY_VERSION-$build_id"
+
+        cd $build_dir
+        docker build -f $cpu_dockerfile --build-arg py_version=$CPU_PY_VERSION --build-arg framework_installable=$cpu_fw_binary -t $PREPROD_IMAGE:$CPU_TAG .
+        cd ../../../
+
+      # build gpu image
+      - |
```

Comment:
it's all pretty repetitive. we should add a build script to the project that we can just call with different args. ok with leaving that for future work though.

Reply:
Yep good idea. I did not think too much because these lines are copied from TC. I will add this in backlog.